components/atoms/details2/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * Components - Details2
+ */
+
+import { useEffect, useRef, useState } from 'react'
+
+// UI
+import { DetailsPropTypes, DetailsDefaultProps } from './props'
+import {
+  DetailsSubscriber,
+  Icon,
+  MessageNames,
+  SIZE,
+  Text,
+  useComponentCommunication
+} from 'components'
+
+// Style
+import styled from 'styled-components'
+
+export const Details2 = ({
+  animationDuration,
+  children,
+  content,
+  context,
+  endActionComponent,
+  icon,
+  open,
+  startActionComponent,
+  title,
+  toolbar,
+  uniqueId,
+  unmountContentOnClose,
+  ...props
+}) => {
+  const [IsOpen, setIsOpen] = useState(open)
+  const [ChildrenMounted, setChildrenMounted] = useState(unmountContentOnClose ? open : true)
+  const [ContentHeight, setContentHeight] = useState(0)
+
+  const contentRef = useRef(null)
+
+  useEffect(() => {
+    if (unmountContentOnClose) {
+      IsOpen
+        ? setChildrenMounted(() => true)
+        : setTimeout(() => {
+            window &&
+              window.requestAnimationFrame(() => {
+                setChildrenMounted(() => false)
+              })
+          }, animationDuration ?? 300)
```

Comment:
It would be good if we add a props to enable/disable animation, like `animation`.